raiden/tests/mocked/test_matrix_transport.py

```diff
@@ -161,13 +197,70 @@ def test_sending_nonstring_body(  # pylint: disable=unused-argument
 def test_processing_invalid_message_json(  # pylint: disable=unused-argument
     mock_matrix, skip_userid_validation, message_input
 ):
-    room, event = make_message(overwrite_data=message_input)
+    room, event = make_message_text(overwrite_data=message_input)
     assert not mock_matrix._handle_sync_messages([(room, [event])])
 
 
 def test_processing_invalid_message_type_json(  # pylint: disable=unused-argument
     mock_matrix, skip_userid_validation
 ):
     invalid_message = '{"_type": "NonExistentMessage", "is": 3, "not_valid": 5}'
-    room, event = make_message(overwrite_data=invalid_message)
+    room, event = make_message_text(overwrite_data=invalid_message)
     assert not mock_matrix._handle_sync_messages([(room, [event])])
+
+
+@pytest.mark.parametrize("retry_interval", [0.01])
+def test_retry_queue_send_only_once(mock_matrix, record_sent_messages):
```

Comment:
I think I understand the problem now:

- `_check_and_send` is called when a new message is added to the queue
- this will send the message once
- The `RetryQueue.run` will wait on `_notify_event`, once the timeout happens it will retry by calling `_check_and_send`.
- Since a timeout happened, a retry should be done, and the message is added to the list of "messages to resent" before checking if the message was removed. Therefore, it will send delivered messages again.

Please, improve the name of this test and add a docstring. Suggestions

Test name: `test_retry_queue_must_not_resent_delivered_messages`
Docstring:
```
RetryQueue used to retry messages even if the message was removed from the queue.
This regression test makes sure that if a message has been delivered, it will not be retried. 
```

Reply:
> Therefore, it will send delivered messages again.

Not `Delivered` but every retryable message can potentially be resent (it they're retry timeout has expired).